File: c#/source/List Folders/List Folders/Includes/Tree/DirNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ListFolders.Includes.Tree {
  class DirNode : TreeNode {

    public List<TreeNode> children;

    public DirNode(String text, List<TreeNode> children) : base(text) {
      this.children = children;
    }

  }
}

File: java/source-eclipse/ListFolders/src/listfolders/includes/tree/DirNode.java
package listfolders.includes.tree;
import java.util.ArrayList;

/*
 * Directory node
 */
public class DirNode extends TreeNode{
  
  public ArrayList<TreeNode> children;
  
  public DirNode(String text, ArrayList<TreeNode> children) {
    super(text);
    this.children = children;
  }

}
